File: api/Models/ContactCustomField.cs
using api.Models.Enums;

namespace api.Models;

public class ContactCustomField
{
	public int Id { get; set; }
	
	public string FieldName { get; set; }

	public string FieldValue { get; set; }

	public ContactCustomFieldTypes FieldType { get; set; }

	public int ContactId { get; set; }

	public Contact Contact { get; set; } = null!;
}

File: frontend-contacts/src/requestTypes/ContactCustomField.ts
import hexoid from 'hexoid';
import type { ContactCustomFieldTypes } from "@/enums/ContactCustomFieldTypes"

export class ContactCustomField {
	id: number
	internalId: string
	fieldName: string
	fieldValue: string
	fieldType: ContactCustomFieldTypes

	constructor(id: number, fieldName: string, fieldValue: string, fieldType: ContactCustomFieldTypes) {
		this.internalId = hexoid(32)()
		this.id = id
		this.fieldName = fieldName
		this.fieldType = fieldType
		this.fieldValue = fieldValue
	}
}